### PR TITLE
copy & import: in case of raw+jpeg make sure they keep the same name

### LINKED ISSUE #10085: No grouping of JPEG and RAW after "Copy and Import"
In Darktable v3.6.1 (at least on Ubuntu 20.04), after executing "copy & import..." on pictures which with double file, i.e., JPEG + RAW, the "collapse grouped images" function in the Lighttable does not work anymore, meaning that corresponding JPEGs and RAWs are not grouped anymore.

I guess that the reason of this issue is that on "copy & import...", each original file gets a different name with a progressive number, so corresponding JPEGs and RAWs get different filenames, losing the property to refer to the same picture. Example: if the original files are A.JPEG and A.RAW, then "copy & import..." will turn them into 0001.JPG and 0002.RAW

Notice that, if I just "add to library..." the pictures (so no copying and renaming), then the automatic grouping of the corresponding JPEGs and RAWs works as expected and each shot appears once in the lighttable.

Is the behavior of no grouping after "copy & import..." intended to be like that?

Best,

Emanuele

### PR DESCRIPTION
**Pull request by phweyland**
fixes #10085

EDIT: if this fix is OK, I'll make the same treatment in dt_camctl_import() (with gphoto2)

### CODE REVIEW

File: src/control/jobs/control_jobs.c [hunk 1]
```diff
@@ -2060,7 +2060,38 @@ void dt_control_write_sidecar_files()
                                                           FALSE));
 }
 
+static gboolean _same_basefilename(const char *filename, const char *prev_filename)
```
Comment (TurboGit) [inline src/control/jobs/control_jobs.c:2063]: I would put that in utility.h and renamed to : `dt_has_same_base_filename()`

And I will make use of `g_path_get_basename)` to simplify the implementation.

File: src/control/jobs/control_jobs.c [hunk 2]
```diff
@@ -2060,7 +2060,38 @@ void dt_control_write_sidecar_files()
...
+  const int prev_length = prev_dot - prev_filename;
+  if(length != prev_length)
+    return FALSE;
+  for(int i = length - 1; i > 0; i--)
+    if(filename[i] != prev_filename[i])
+      return FALSE;
+  return TRUE;
+}
+
+static char *_set_extension(const char *prev_output, const char *filename)
```
Comment (TurboGit) [inline src/control/jobs/control_jobs.c:2080]: Same here, I'll move that into utility.h.

Also I would use `g_malloc()`.

File: src/control/jobs/control_jobs.c [hunk 3]
```diff
@@ -2060,7 +2060,38 @@ void dt_control_write_sidecar_files()
...
+{
+  const char *dot = strrchr(filename, '.');
+  const char *prev_dot = strrchr(prev_output, '.');
+  const int name_lgth = prev_dot - prev_output;
+  const int ext_lgth = strlen(dot);
+  char *output = malloc(name_lgth + ext_lgth + 1);
+  memcpy(output, prev_output, name_lgth);
+  memcpy(&output[name_lgth], &filename[strlen(filename) - ext_lgth], ext_lgth);
```
Comment (TurboGit) [inline src/control/jobs/control_jobs.c:2088]: `... ext_lgth + 1);`

Would copy the ending \0 and then you don't need following line?

**Review by TurboGit — CHANGES_REQUESTED** [review]
Some minor changes. TIA.

File: src/control/jobs/control_jobs.c (hunk 1 above)
Comment (phweyland) [reply src/control/jobs/control_jobs.c:2063]: > I would put that in utility.h and renamed to : dt_has_same_base_filename()

I'll do it.

> And I will make use of g_path_get_basename) to simplify the implementation.

g_path_get_basename() returns the last component of the filename. I don't see how I can use this here.

File: src/control/jobs/control_jobs.c (hunk 3 above)
Comment (phweyland) [reply src/control/jobs/control_jobs.c:2088]: Yes.

File: src/control/jobs/control_jobs.c (hunk 2 above)
Comment (phweyland) [reply src/control/jobs/control_jobs.c]: done

**Comment by phweyland** [comment]
Requested changes done. Thanks !

File: src/control/jobs/control_jobs.c (hunk 1 above)
Comment (TurboGit) [reply src/control/jobs/control_jobs.c]: I see, this is not checking **basename** but **full pathname** without extension. The routine name is confusing then.

Comment (TurboGit) [reply src/control/jobs/control_jobs.c]: I see that you have renamed it to: `dt_has_same_path_basename()`, so this convey better the implementation.

**Review by TurboGit — APPROVED** [review]
Looks ok to me now. Thanks.

**Comment by wpferguson** [comment]
@phweyland do we need to update film_jobs.c also?  Changes here look good to me.

**Comment by phweyland** [comment]
> do we need to update film_jobs.c also?

If I'm not mistaken film_job.c is only used with darktable-cli now and it imports the images where and as they are. So there is now issue in that context. Thanks.
Remains the camera import now.

**Pull request merged**